### PR TITLE
Allow multiple geography files to be specified

### PR DESCRIPTION
**Pull request by phynes-sensiblecode**

### CODE REVIEW

File: bin/ons_csv_to_ctb_json_geo.py [hunk 1]
```diff
@@ -32,13 +47,15 @@ def read_geo_cats(filename):
        and LAD22cd.
      - If multiple lines in the file refer to the same category then the names must be consistent
        on all lines.
+     - The column names are handled in a cases insensitive manner.
```
Comment (mikejamesthompson) [inline bin/ons_csv_to_ctb_json_geo.py:50]: `cases insensitive` -> `case-insensitive`

File: bin/ons_csv_to_ctb_json_main.py [hunk 2]
```diff
@@ -76,11 +77,21 @@ def main():
...
-                             'variables')
+    # Keeping the -g parameter maintains backwards compatibility to case where only a single
+    # geography file was supported. The list of files could be quite long, so also adding
+    # the -d option.
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument('-g', '--geography-files',
+                       type=str,
+                       required=False,
+                       help='Names of CSV files containing category codes and names for '
+                            'geographic variables, supplied as a comma separated list')
```
Comment (mikejamesthompson) [inline bin/ons_csv_to_ctb_json_main.py:88]: Should be:
> comma-separated

File: bin/ons_csv_to_ctb_json_main.py [hunk 3]
```diff
@@ -396,7 +419,11 @@ def build_ctb_service_metadata(metadata_version_number, build_time, args):
                 'created': build_time,
                 'dataset_filter': args.dataset_filter if args.dataset_filter else None,
                 'best_effort': str(args.best_effort),
-                'geography_file': geography_file(args) if args.geography_file else None,
+                # The geography files would be better as a list. However, that would involve an
+                # API change and could potentially requires updates to existing software using the
```
Comment (mikejamesthompson) [inline bin/ons_csv_to_ctb_json_main.py:423]: `could potentially requires updates to` -> `could impact`

File: bin/ons_csv_to_ctb_json_main.py [hunk 4]
```diff
@@ -410,9 +437,9 @@ def build_ctb_service_metadata(metadata_version_number, build_time, args):
     return [service_metadata.english(), service_metadata.welsh()]
 
 
-def geography_file(args):
-    """Return the basename of the geography file."""
-    return f'{os.path.basename(args.geography_file)}' if args.geography_file else ""
+def basename_string(filenames):
+    """Return a comma separate list of basenames of a list of files."""
```
Comment (mikejamesthompson) [inline bin/ons_csv_to_ctb_json_main.py:441]: `Return a comma separate list ...` -> `Return a string containing a comma-separated list ...`

File: CENSUS_METADATA.md [hunk 5]
```diff
@@ -111,7 +111,7 @@ It provides some information about the build process, where the CSV files were c
 | `created` | `String!` | Time in ISO 8601 format that the Python CSV to JSON script was executed | |
 | `best_effort` | `String!` | `"TRUE"` if the `--best-effort` flag was used when running the script, else `"FALSE"` | |
 | `dataset_filter` | `String` | Contains the values specified with the `--dataset-filter` command line parameter, if it was used when running the script | |
-| `geography_file` | `String` | Name of the geography file used to source category labels for geographic variables if supplied on the command line | |
+| `geography_file` | `String` | Comma separated list of the names of geography files used to source category labels for geographic variables if supplied on the command line | |
```
Comment (mikejamesthompson) [inline CENSUS_METADATA.md:114]: `Comma separated` -> `Comma-separated`

**Review by mikejamesthompson — APPROVED** [review]
I've skimmed the code and docs changes and spotted a few tiny text changes.

I also built and tested the metadata in a few steps. First loading one geo file with `-g`, then two with `-g` and then the whole directory and confirmed the content appears for the `catLabels` field at each stage with the following query:

```graphql
{
  dataset(name:"UR", lang: "en") {
    name label
    vars(names:["sex", "rgn", "wpc", "nhser"]) {
      name label
      catLabels
      meta {
        Version
        Cantabular_Public_Flag
        ONS_Variable {
          Variable_Mnemonic
        }
      }
    }
  }
}
```

All looks good to me 👍

File: bin/ons_csv_to_ctb_json_geo.py (hunk 1 above)
Comment (phynes-sensiblecode) [reply bin/ons_csv_to_ctb_json_geo.py:50]: OK

File: bin/ons_csv_to_ctb_json_main.py (hunk 2 above)
Comment (phynes-sensiblecode) [reply bin/ons_csv_to_ctb_json_main.py:88]: OK

File: bin/ons_csv_to_ctb_json_main.py (hunk 4 above)
Comment (phynes-sensiblecode) [reply bin/ons_csv_to_ctb_json_main.py:441]: OK

File: CENSUS_METADATA.md (hunk 5 above)
Comment (phynes-sensiblecode) [reply CENSUS_METADATA.md:114]: OK

**Pull request merged**